Use flyway for database change management #35

mp911de opened the pull request:
Add liquibase dependency. Replace auto-ddl with liquibase to apply only missing database changesets. Create multiple changesets. Adjust table and column names for user to email mapping. Move dev-related properties to dev profile.

mp911de committed:
Use flyway for database change management.

Add flyway dependency. Replace auto-ddl with flyway to apply only missing database changesets. Adjust table and column names for user to email mapping. Move dev-related properties to dev profile.

gh-35

mp911de committed:
Add CloudFoundry database connectors

gh-35

rwinch commented on build.gradle:
```diff
@@ -46,6 +46,8 @@ dependencies {
 	compile('net.sourceforge.nekohtml:nekohtml')
 	compile('org.projectlombok:lombok:1.16.6')
+	compile('org.flywaydb:flyway-core:4.0')
+	compile('mysql:mysql-connector-java:5.1.38')
```
Is `mysql:mysql-connector-java` necessary still?

mp911de replied:
Yes, otherwise starting the app will fail:

```
2016-04-19T09:28:33.63+0200 [APP/0]      ERR Caused by: org.cloudfoundry.reconfiguration.org.springframework.cloud.CloudException: No suitable database driver found for githubcla-db service 
2016-04-19T09:28:33.63+0200 [APP/0]      ERR    at org.cloudfoundry.reconfiguration.org.springframework.cloud.service.relational.DataSourceCreator.getDriverClassName(DataSourceCreator.java:81)
2016-04-19T09:28:33.63+0200 [APP/0]      ERR    at org.cloudfoundry.reconfiguration.org.springframework.cloud.service.relational.DataSourceCreator.create(DataSourceCreator.java:50)
2016-04-19T09:28:33.63+0200 [APP/0]      ERR    ... 24 more
```

rwinch commented on src/main/resources/db/migration/V1_1__create-tables.sql:
```diff
@@ -0,0 +1,90 @@
+ALTER TABLE user_email
+  ADD CONSTRAINT FK_USER_EMAIL
+FOREIGN KEY (github_login) REFERENCES USER (github_login);
+
```
It would probably be good to add indexes for the attributes that we query / join by.

mp911de replied:
Done.

mp911de committed:
Add indexes to queried/join-by attributes.

gh-35

mp911de committed:
Use h2 database on PWS.

Disable JavaBuildPack auto-reconfiguration and enable local profile to use h2 on PWS.

gh-35

mp911de committed:
Add non-null constraints to version columns and CLA FKs

Fixes gh-37

mp911de committed:
Remove test code.

mp911de committed:
Lookup CLAs instead of creating those.

Flyway initializes the database with the required data so it's no longer necessary to create own CLA records.

rwinch commented on src/main/resources/db/migration/V1_1__create-tables.sql:
```diff
@@ -0,0 +1,103 @@
+);
+
+
+CREATE TABLE USER (
```
recently ADMIn and admin_access_requested were marked as transient. Can we remove these properties?

mp911de replied:
Done.

mp911de committed:
Remove ADMIN and admin_access_requested columns.
